Inc/DirectXMath.h

```diff
@@ -141,7 +163,27 @@
 #endif
 #endif // !_XM_NO_INTRINSICS_
 
+#ifdef _WIN32
 #include <sal.h>
+#else
+#define _Analysis_assume_(x)
```

Comment:
My general answer is to have a sal.h in your system paths that does this... Avoids cluttering up the header.

Reply:
You have no possibility to have sal.h on Linux.

Reply:
Actually, there's an open source version of SAL.H under the MIT license you can use [here](https://github.com/dotnet/corert/blob/master/src/Native/inc/unix/sal.h)

Reply:
Can you include the open source sal.h, and have an ifdef, something like

```
#ifndef MSVC
#include "opensoruce_sal.h"
#else
#include "sal.h"
#endif
```

???

Reply:
Yes, but it is difficult to tell users that they should take header file somewhere from another project. 

Ideally, the library without dependencies should be without dependencies :)

Reply:
Updated my prev comment, I guess it wasn't super clear. I meant to say package the open source sal.h with the DirectXMath repo, and use it if we're not compiling with MSVC.

Reply:
sal.h is already there for clang for Windows as well.

Reply:
Picks up the sal.h, but the compiler is not super happy...

```
/Users/andy/src/mechanica/extern/DirectXMath/Inc/DirectXMath.h:1880:25: warning: calling convention '__fastcall' ignored for this target [-Wignored-attributes]
    inline XMVECTOR     XM_CALLCONV     XMVectorRotateRight(FXMVECTOR V)
                        ^
/Users/andy/src/mechanica/extern/DirectXMath/Inc/DirectXMath.h:29:21: note: expanded from macro 'XM_CALLCONV'
#define XM_CALLCONV __fastcall
                    ^
/Users/andy/src/mechanica/extern/DirectXMath/Inc/DirectXMath.h:1887:25: warning: calling convention '__fastcall' ignored for this target [-Wignored-attributes]
    inline XMVECTOR     XM_CALLCONV     XMVectorInsert(FXMVECTOR VD, FXMVECTOR VS)
                        ^
/Users/andy/src/mechanica/extern/DirectXMath/Inc/DirectXMath.h:29:21: note: expanded from macro 'XM_CALLCONV'
#define XM_CALLCONV __fastcall
                    ^
/Users/andy/src/mechanica/extern/DirectXMath/Inc/DirectXMath.h:1911:1: error: '__declspec' attributes are not enabled; use '-fdeclspec' or '-fms-extensions' to enable support for __declspec attributes
XMGLOBALCONST XMVECTORF32 g_XMSinCoefficients0      = { { { -0.16666667f, +0.0083333310f, -0.00019840874f, +2.7525562e-06f
```

Reply:
Looking ad adding the '-fdeclspec' or '-fms-extensions' options now...

Reply:
Adding ``-fms-extensions`` will fix the issues for x64/x86.

For ARM/ARM64, you need some additional #defines for missing intrinsics because all the ``_ex`` versions which take an alignment are unsupported on non-Windows ARM platforms. These can be easily mapped to their non-ex versions.

Reply:
OK, compiles, still has tons of ```warning: calling convention '__fastcall' ignored for this target``` warnings. 

Looking into this now...

Reply:
The easiest way to get rid off  these warnings is to replace

```
#elif !defined(__GNUC__) 
#define XM_CALLCONV __fastcall
```

with

```
#elif !defined(__GNUC__) && !defined(__APPLE__ )
#define XM_CALLCONV __fastcall
```

Reply:
What *seems* to work fine, is :

```
#if defined(__APPLE__) 
#define __fastcall
#endif
#include "DirectXMath.h"
#endif
```

No warnings, lets test some functionality....